File: examples/simp/Python/predictor/predictor4.py
# Import the library
import LinkPredPython as lpp
# Create a prtedictor object
p = lpp.Predictor();
# Load network from file
p.loadnet("Zakarays_Karate_Club.edges");
esv = lpp.EdgeScoreVec();
es = lpp.EdgeScore();
es.i = "1";
es.j = "34";
esv.push_back(es);
es.i = "26";
es.j = "34";
esv.push_back(es);
p.predKAB(esv);
# Print the scores
for es in esv:
	print(es.i + "\t" + es.j + "\t" + "{:.4f}".format(es.score));

File: examples/simp/CPP/predictor/predictor4.cpp
#include <linkpred.hpp>
#include <iostream>

using namespace LinkPred::Simp;

int main() {
	Predictor p; // Create a prtedictor object
	p.loadnet("Zakarays_Karate_Club.edges"); // Load network from file
	std::vector<EdgeScore> esv = { { "1", "34" }, { "26", "34" } };
	p.predKAB(esv);
	for (auto it = esv.begin(); it != esv.end(); ++it) {
		// Print the scores
		std::cout << it->i << "\t" << it->j << "\t" << it->score << std::endl;
	}
	return 0;
}
